include/zephyr/dt-bindings/adc/stm32_adc.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2023 Linaro Ltd.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_ADC_STM32_ADC_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_ADC_STM32_ADC_H_
+
+#define STM32_ADC_RES_RES_MASK    0x15U
+#define STM32_ADC_RES_RES_SHIFT   0U
+#define STM32_ADC_RES_VAL_MASK    0x7U
+#define STM32_ADC_RES_VAL_SHIFT   4U
+#define STM32_ADC_RES_OFF_MASK    0x31U
+#define STM32_ADC_RES_OFF_SHIFT   6U
```

Comment:
```
#define STM32_ADC_RES_RES_MASK    0x1FU
#define STM32_ADC_RES_RES_SHIFT   0U
#define STM32_ADC_RES_VAL_MASK    0x7U
#define STM32_ADC_RES_VAL_SHIFT   5U
#define STM32_ADC_RES_OFF_MASK    0x1FU
#define STM32_ADC_RES_OFF_SHIFT   8U
```
Some masks are wrong, and we need to take more space because of some STM32 series